### ISSUE TITLE
Resulting equation

### ISSUE THREAD

**Issue by oscargus** [2pi/karnaugh-map#5]
It is sometimes useful to write the resulting equation below the Karnaugh map. This can of course be done externally, but on the other hand, all the information required is already entered (implicants and variable names).

As an added bonus: each term can be colored as the corresponding implicant.

Second bonus: arrow from each term to the corresponding implicant.

**Comment by oscargus** [comment]
I realized that I can probably add some sort of caption support myself. There will be a PR eventually for this (just adding arbitrary content below the Karnaugh map).

**Comment by 2pi** [comment]
> write the resulting equation below the Karnaugh map /**/ all the information required is already entered

I've thought about this feature too, however it would require some latexing! Regardless it would be a killer feature! :+1:

> each term can be colored as the corresponding implicant

Great idea! :1st_place_medal: 

> arrow from each term to the corresponding implicant

I think this will be a little to much. How would the arrows go when the map is really complicated...? Maybe an API exposing positions of different parts of the map would be better. This will essentially solve issues such as this, #3 and #4.

>  realized that I can probably add some sort of caption support myself

I usually put the karnaugh map inside of a figure and then use the figure caption

**Comment by 2pi** [comment]
Just to update, I've a working, but not polished, implementation of this I did a few years ago for when marking minterms but not maxterms in latex. It is kind of complex, but we'll have to see. Anyway I've added support for specifying labels separately as that is required in order to generate the equation properly. It also supports coloring the output function which is kind of nice.

**Comment by 2pi** [comment]
Moving this to v2.1 so we can publish the other things that is cooking in this repo at the moment.

**Comment by oscargus** [comment]
Great that there are some progress with karnaugh-map in general!

An option, not very compact though, is to use SymPy. I've been thinking about using karnaugh-map either in SymPy or in a spin-off so that K-maps can be shown in rendered form in an IPython console (or Jupyter Notebook).

**Comment by 2pi** [comment]
Yeah, definitely! I'm particularly happy that I did get around putting up some code that I've been working on for quite some time.

Regarding SymPy; I'm not entirely sure what you are proposing. Would you mind expanding?

**Comment by oscargus** [comment]
Nice with some improvements!

SymPy can do Boolean optimization and produce LaTeX for the resulting equation. But it will be a quite large dependency for this...

(One day, I will add things, either to SymPy or a separate extension package, to be able to generate LaTeX for the k-maps automatically including resulting implicants etc. Also, strictly, one would probably like the LaTeX to use a more "Boolean" approach to the equations, +, '/overbar, etc rather than \vee etc.)

**Comment by 2pi** [comment]
Aha, yeah that is quite a big dependency. ~Another option would be `sage` but that is also quite large.~ On the positive side there is only need for a 6 variable implementation as anything more can't be drawn with this package ;)
